executor/executor.go

```diff
@@ -1527,6 +1536,9 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 		MemTracker:  memory.NewTracker(stringutil.MemoizeStr(s.Text), vars.MemQuotaQuery),
 		DiskTracker: disk.NewTracker(stringutil.MemoizeStr(s.Text), -1),
 	}
+	if config.GetGlobalConfig().OOMUseTmpStorage && GlobalDiskUsageTracker != nil {
+		sc.DiskTracker.AttachTo(GlobalDiskUsageTracker)
```

Comment:
The `sc.DiskTracker` of the previous statement should be detached firstly, or the children of `GlobalDiskUsageTracker` cannot be garbage collected.